spec/decorators/corporate_decorator_spec.rb

```diff
@@ -0,0 +1,39 @@
+RSpec.describe CorporateDecorator do
+  include Draper::ViewHelpers
+
+  subject(:decorator) { described_class.decorate(content_item) }
+
+  let(:content_item) { instance_double(Core::Article, id: 'bob', slug: 'syndicated') }
+
+  describe '#extra_content?' do
+    context 'when article has partial' do
+      before do
+        expect_any_instance_of(ActionView::LookupContext).to receive(:template_exists?)
```

Comment:
This feels cumbersome, can this behaviour not be observed or triggered by other means? Hooking into rails internals is generally a bad idea :tm: